### PR TITLE
Delivery date da

### LINKED ISSUE #21: BE - Delivery Date Adjustment
Tasks:
- [x] When a subscription is added, set the recurring shipment date to subscription date for future months.
- [x] Each shipment will have a & delivery date.
- [x] User can edit delivery date for each shipment.
- [x] Create mutation to edit shipment and a resolver.

### PR DESCRIPTION
**Pull request by atkinson-daniel**
# PR Description
Please include a summary of the change:

:bar_chart: Complete BE Delivery Date Update functionality. Closes #21 

If it is a new feature, describe the feature and the iteration of the project spec it addresses.

1) Add a hook to the subscription model to create a year (12 months) of shipments for the user's subscription
2) Create an update deliveryDate mutation to allow the user to update a deliveryDate
3) Create a getAll query to get all delivery dates

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/api/src/modules/shipments/mutations.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// Delivery Date Update
+export const deliveryUpdate = {
+  type: ShipmentType,
+  args: {
+    id: {
+      name: 'id',
+      type: GraphQLInt
+    },
+    deliveryDate: {
```
Comment (zachholcomb) [inline code/api/src/modules/shipments/mutations.js:28]: this was a good call to only allow for the date to be edited

File: code/api/src/modules/subscription/model.js [hunk 2]
```diff
@@ -9,12 +11,26 @@ module.exports = function(sequelize, DataTypes) {
...
     }
-  })
+  }, {
+      hooks: {
+        afterCreate: async (subscription) => {
+          let user = subscription.dataValues.userId;
+          let subscriptionId = subscription.dataValues.id;
+          let date = new Date(new Date().getTime()+(5*24*60*60*1000))
+
+          for(let i = 1; i < 13; i++){
```
Comment (zachholcomb) [inline code/api/src/modules/subscription/model.js:21]: I think this is a good solution for the time being, we can circle back if we have time!

File: code/api/tests/subscription.test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    server.use(
+      '/',
+      graphqlHTTP(request => ({
+        schema: schema,
+        graphiql: false
+      })
+    ))
+  });
+
+  it('returns all subscriptions', async () => {
```
Comment (zachholcomb) [inline code/api/tests/subscription.test.js:21]: With the seeds on master we'll be able to test this a little better, besides the auth queries

File: code/api/src/modules/shipments/mutations.js (hunk 1 above)
Comment (infamouskeyduster) [reply code/api/src/modules/shipments/mutations.js:28]: I agree with that! No need to alter any of the other data coming through.

**Comment by zachholcomb** [comment]
Looks good to me!

**Review by infamouskeyduster — APPROVED** [review]
This is John Watterson, and I approve these changes ('said in a stoic politician ad campaign voice').

**Pull request merged**